components/powersensor/server_test.go

```diff
@@ -0,0 +1,140 @@
+package powersensor_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	pb "go.viam.com/api/component/powersensor/v1"
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/components/powersensor"
+	"go.viam.com/rdk/components/sensor"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/testutils/inject"
+)
+
+var (
+	workingPowerSensorName = "ps1"
+	failingPowerSensorName = "ps2"
+	missingPowerSensorName = "ps3"
+	errVoltageFailed       = errors.New("can't get voltage")
+	errCurrentFailed       = errors.New("can't get current")
+	errPowerFailed         = errors.New("can't get power")
+	errPowerSensorNotFound = errors.New("not found")
+)
+
+func newServer() (pb.PowerSensorServiceServer, *inject.PowerSensor, *inject.PowerSensor, error) {
+	workingPowerSensor := &inject.PowerSensor{}
+	failingPowerSensor := &inject.PowerSensor{}
+	powerSensors := map[resource.Name]powersensor.PowerSensor{
+		powersensor.Named(workingPowerSensorName): workingPowerSensor,
+		powersensor.Named(failingPowerSensorName): failingPowerSensor,
+	}
+
+	powerSensorSvc, err := resource.NewAPIResourceCollection(sensor.API, powerSensors)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return powersensor.NewRPCServiceServer(powerSensorSvc).(pb.PowerSensorServiceServer), workingPowerSensor, failingPowerSensor, nil
+}
+
+//nolint:dupl
```

Comment:
I infer from this comment that the linter thinks there's lots of duplicate code in here. Can it be made into a helper function that is called multiple times, rather than copied-and-pasted code? "No" and even "yes, but it's not worth the hassle so I'm not going to do it" are plausible responses, but I figured I'd ask.

Reply:
might be possible but it would have to be multiple small helper functions so don't think its worth doing